lib/puppet/type/service.rb

```diff
@@ -237,6 +239,16 @@ def sync
       desc "Specify a command to config a service, or a path to a manifest to do so."
     end
 
+    newparam(:timeout, :required_features => :configurable_timeout) do
+      desc "Specify a timeout (in seconds) for service change operations."
```

Comment:
Would probably be clearer if this was written as:
```
An (optional) timeout that specifies how long Puppet should wait when syncing the ensure property.
```

May be worth mentioning that this is not strictly enforced when, for example, we're waiting on a pending state because we reset the elapsed time everytime the checkpoint flag is incremented.

Reply:
Yeah, I can do that. normally I just call it a 'minimum timeout' to indicate puppet may wait longer if the service is responsive but long to perform the operation.

Reply:
Discussed in hipchat:

Since the feature is type (and not provider) specific it's possible future providers will use timeout in other ways, so we are going to leave out mentioning ensure